multi_partner_learning.py

```diff
@@ -458,27 +447,47 @@ def prepare_aggregation_weights(self):
     def aggregate_model_weights(self):
         """Aggregate model weights from the list of models, with a weighted average"""
 
-        weights_per_layer = list(zip(*self.models_weights_list))
-        new_weights = list()
+        # SKL models weigths are tuples while Keras model's weights are list
+        if type(self.models_weights_list[0]) is tuple:  # Check weights type for aggregation
+            # Unpack values
+            coefs = [weights[0] for weights in self.models_weights_list]
+            intercepts = [weights[1] for weights in self.models_weights_list]
 
-        for weights_for_layer in weights_per_layer:
-            avg_weights_for_layer = np.average(
-                np.array(weights_for_layer), axis=0, weights=self.aggregation_weights
-            )
-            new_weights.append(list(avg_weights_for_layer))
+            agg_coef = np.average(np.array(coefs), axis=0, weights=self.aggregation_weights)
+            agg_intercepts = np.average(np.array(intercepts), axis=0, weights=self.aggregation_weights)
+
+            new_weights = (agg_coef, agg_intercepts)
+        else:
+            weights_per_layer = list(zip(*self.models_weights_list))
+            new_weights = list()
+
+            for weights_for_layer in weights_per_layer:
+                avg_weights_for_layer = np.average(
+                    np.array(weights_for_layer), axis=0, weights=self.aggregation_weights
+                )
+                new_weights.append(list(avg_weights_for_layer))
 
         return new_weights
-    
+
+    def save_model_for_partner(self, model, partner_index):
+        """save a model with weight"""
+        if isinstance(model, type(LogisticRegression())):
+            self.models_weights_list[partner_index] = (model.coef_, model.intercept_)
+        else:
+            self.models_weights_list[partner_index] = model.get_weights()
 
     def build_model_from_weights(self, new_weights):
         """Generate a new model initialized with weights passed as arguments"""
 
         new_model = self.generate_new_model()
-        new_model.set_weights(new_weights)
 
+        if isinstance(new_model, type(LogisticRegression())):
+            new_model.coef_, new_model.intercept_ = new_weights
+            new_model.n_iter_ = 0
```

Comment:
Why do we have to do that ?

Reply:
Hmm, good question, with a quick search I didn't find anything about putting `n_iter` to `0`.

Reply:
I ran test without it and can't see an impact. I commented it out for the moment (and asked @aygalic if he can clarify).

Reply:
And finally I removed it after discussing with @aygalic . We should be all good